cmd/updatetree/main.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main contains the implementation and entry point for the updatetree
+// command.
+//
+// Example usage:
+// $ ./updatetree --admin_server=host:port --tree_id=123456789 --tree_state=FROZEN
+//
+// The output is minimal to allow for easy usage in automated scripts.
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/proto"
+	"github.com/google/trillian"
+	"google.golang.org/genproto/protobuf/field_mask"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	adminServerAddr = flag.String("admin_server", "", "Address of the gRPC Trillian Admin Server (host:port)")
+	rpcDeadline     = flag.Duration("rpc_deadline", time.Second*10, "Deadline for RPC requests")
+	treeID          = flag.Int64("tree_id", 0, "The ID of the tree to be set updated")
+	treeState       = flag.String("tree_state", "", "If set the tree state will be updated")
+)
+
+func updateTree(ctx context.Context) (*trillian.Tree, error) {
```

Comment:
Personally, I prefer restricting flag processing to `main` and so would have `(ctx, server, treeID, treeState)` parameters here -- it would also mean the unit tests were more obviously function-based and didn't need flagsavers.

Reply:
If I'd written it I would have done that. I'll TODO both programs to clean this up.